metapool/calculate_coefficients.py

```diff
@@ -0,0 +1,147 @@
+    # Strip leading X in sample names which are automatically removed by R
+    # table reading.
+    metadata_pools["sample_name_r"] = metadata_pools["sample_name_r"].str[1:]
+
+    # Parsing the files
+    # Calculate the total number of reads aligned
+    # to the plasmid sequences for each sample
+    table_synthetic_hit_totals = table_synthetic_hits.sum()
+    table_synthetic_hit_totals = pd.DataFrame(
+        data={"sample_name_r": table_synthetic_hit_totals.index.to_list(),
+              "table_synthetic_hit_totals":
+                  table_synthetic_hit_totals.values.tolist()},
+        index=None)
+
+    # Convert feature-table to long-form and merge with total number of hits to
+    # plasmid sequences
+    table_synthetic_hits_long_with_totals_beta = pd.melt(
+        table_synthetic_hits.reset_index(), id_vars="OTUID")
+    table_synthetic_hits_long_with_totals_beta.columns = ["plasmid_id",
+                                                          "sample_name_r",
+                                                          "count"]
+    table_synthetic_hits_long_with_totals = \
+        table_synthetic_hits_long_with_totals_beta.merge(
+            table_synthetic_hit_totals, on="sample_name_r")
+
+    # Merge updated feature-table with sample-pool information
+    table_synthetic_hits_long_with_totals_and_pools = \
+        table_synthetic_hits_long_with_totals.merge(
+            metadata_pools, on="sample_name_r")
+
+    # Aggregate across lanes (not needed)
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes = \
+        table_synthetic_hits_long_with_totals_and_pools
+
+    # Calculate counts per million
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+        "counts_per_million"] = (
+            table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+                "count"] /
+            table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+                "read_count_total"]
+        ) * 1000000
+
+    # Calculate percentages
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes["percentage"] = (
+       table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+          "count"] /
+       table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+          "read_count_total"]
+    ) / 100
+
+    # Calculate log10 of counts per million
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+        "counts_per_million_log10"] = np.log10(
+        table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+            "counts_per_million"])
+
+    # Merge table with plasmid dilution information
+    table_synthetic_hits_with_dilutions = \
+        table_synthetic_hits_long_with_totals_and_pools_all_lanes.merge(
+            dilutions, on=["plasmid_id", "pool"])
+
+    final_table_1 = table_synthetic_hits_with_dilutions
+
+    dilution_log = final_table_1["dilution_id"].copy()
+    # TODO FIXME HACK:  Check that the dilutions we aim for are 5%, .5%, .05%
```

Comment:
Sent an email to @justinshaffer to confirm; however, comment should be removed when solved.